src/Jackett.Common/Indexers/AwesomeHD.cs

```diff
@@ -0,0 +1,188 @@
+            : base("Awesome-HD",
+                description: "An HD tracker",
+                link: "https://awesome-hd.me/",
+                caps: new TorznabCapabilities(),
+                configService: configService,
+                client: c,
+                logger: l,
+                p: ps,
+                configData: new ConfigurationDataPasskey())
+        {
+            Encoding = Encoding.UTF8;
+            Language = "en-us";
+            Type = "private";
+
+            TorznabCaps.SupportsImdbMovieSearch = true;
+            TorznabCaps.SupportsImdbTVSearch = true;
+
+            AddCategoryMapping(1, TorznabCatType.MoviesHD);
+            AddCategoryMapping(2, TorznabCatType.TVHD);
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            if (configData.Passkey.Value.Length != 32)
+                throw new Exception("Invalid Passkey configured. Expected length: 32");
+
+            var releases = await PerformQuery(new TorznabQuery());
+            await ConfigureIfOK(string.Empty, releases.Any(),
+                                () => throw new Exception("Could not find release from this URL."));
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var passkey = configData.Passkey.Value;
+
+            var qc = new NameValueCollection
+            {
+                {"passkey", passkey}
+            };
+
+            if (!string.IsNullOrWhiteSpace(query.ImdbID))
+            {
+                qc.Add("action", "imdbsearch");
+                qc.Add("imdb", query.ImdbID);
+            }
+            else if (!string.IsNullOrWhiteSpace(query.GetQueryString()))
+            {
+                qc.Add("action", "titlesearch");
+                qc.Add("title", query.SearchTerm); // not use query.GetQueryString(), see the season code below
+            }
+            else
+                qc.Add("action", "latestmovies");
+
+            var searchUrl = SiteLink + SearchUrl + "?" + qc.GetQueryString();
+            var results = await RequestStringWithCookies(searchUrl);
+            if (string.IsNullOrWhiteSpace(results.Content))
+                throw new Exception("Empty response. Please, check the Passkey.");
+
+            try
+            {
+                var doc = XDocument.Parse(results.Content);
+
+                var torrents = doc.Descendants("torrent").ToList();
+                if (!torrents.Any())
+                    return releases; // no results
+
+                var authkey = doc.Descendants("authkey").First().Value;
+
+                foreach (var torrent in torrents)
```

Comment:
If `torrents` is empty then the `foreach` won't run, we can simplify this to and get the same effect: 
```suggestion
                var authkey = doc.Descendants("authkey").First().Value;
                foreach (var torrent in doc.Descendants("torrent"))
```

Reply:
The problem was `authkey` is null too in that case, so I did that to avoid more checks.
I improved this part to make it more clear.